src/libs/Navigation/AppNavigator/AuthScreens.js

```diff
@@ -0,0 +1,353 @@
+        }).then(subscribeToReportCommentEvents);
+
+        // Fetch some data we need on initialization
+        NameValuePair.get(CONST.NVP.PRIORITY_MODE, ONYXKEYS.PRIORITY_MODE, 'default');
+        PersonalDetails.fetch();
+        PersonalDetails.fetchTimezone();
+        getBetas();
+        fetchAllReports(true, true);
+        fetchCountryCodeByRequestIP();
+        UnreadIndicatorUpdater.listenForReportChanges();
+
+        // Refresh the personal details, timezone and betas every 30 minutes
+        // There is no pusher event that sends updated personal details data yet
+        // See https://github.com/Expensify/ReactNativeChat/issues/468
+        this.interval = setInterval(() => {
+            if (this.props.network.isOffline) {
+                return;
+            }
+            PersonalDetails.fetch();
+            PersonalDetails.fetchTimezone();
+            getBetas();
+        }, 1000 * 60 * 30);
+
+        Timing.end(CONST.TIMING.HOMEPAGE_INITIAL_RENDER);
+
+        // Listen for the Command+K key being pressed so the focus can be given to the chat switcher
+        KeyboardShortcut.subscribe('K', () => {
+            redirect(ROUTES.SEARCH);
+        }, ['meta'], true);
+    }
+
+    componentWillUnmount() {
+        KeyboardShortcut.unsubscribe('K');
+        NetworkConnection.stopListeningForReconnect();
+        clearInterval(this.interval);
+        this.interval = null;
+    }
+
+    render() {
+        const modalScreenOptions = {
+            headerShown: false,
+            cardStyle: getNavigationModalCardStyle(this.props.isSmallScreenWidth),
+            gestureDirection: 'horizontal',
+        };
+
+        if (this.props.isSmallScreenWidth) {
+            modalScreenOptions.cardStyleInterpolator = CardStyleInterpolators.forScaleFromCenterAndroid;
+        }
+
+        const SettingsModalStack = this.props.responsive
+            ? createCustomModalStackNavigator()
+            : createStackNavigator();
+
+        const NewChatModalStack = this.props.responsive
+            ? createCustomModalStackNavigator()
+            : createStackNavigator();
+
+        const NewGroupModalStack = this.props.responsive
+            ? createCustomModalStackNavigator()
+            : createStackNavigator();
+
+        const SearchModalStack = this.props.responsive
+            ? createCustomModalStackNavigator()
+            : createStackNavigator();
+
+        const ProfileModalStack = this.props.responsive
+            ? createCustomModalStackNavigator()
+            : createStackNavigator();
+
+        const IOURequestModalStack = this.props.responsive
+            ? createCustomModalStackNavigator()
+            : createStackNavigator();
+
+        const IOUBillModalStack = this.props.responsive
+            ? createCustomModalStackNavigator()
+            : createStackNavigator();
+
+        return (
+            <RootStack.Navigator
```

Comment:
It does seem incorrect that you can swipe right to the chat page. Maybe this is natural for iOS, but the concept of navigating forwards to a closed page is not aligned with expected Android navigation.  

I want to expand on this with a few navigation quirks: 

**A)**  It's super jarring to me as an Android user that we have the iOS page slide gesture. Combined with the native Android drag to trigger back button press you get different actions depending on how far from the side of the screen that you begin the gesture. I would suggest the iOS slide is only applied to iOS (_notice the difference between manual slide between pages and the `<` symbol that appears when triggering a back button press_).

**B)** Also, it seems that our navigation stack is not quite working as expected, because when at the top of navigation hierarchy (the NavDrawer), pressing the native back button (or using the gesture) actually takes you deeper into the app by reopening the the 1:1 chat page which has been closed. The Android 'Back' event should always take you 'up a navigation level', or close the app if you are already at the top level page (technically bottom of the nav stack as you pop the pages). As users may have entered our app via another (Gmail for example), they should be able to complete their task, then press back to clear pages one on one until they reach Gmail again. Some apps keep users in their app by inserting their home page between Gmail and their entry page, but in our case we're adding new pages to the stack on top of the current page, which seems even more confusing.

Please let me know if you'd like me to expand further, I'm sure it's a bit tricky to reconcile both native mobile patterns.

https://user-images.githubusercontent.com/10736861/110662316-4c124580-81bd-11eb-88d2-264c9824260b.mp4

Reply:
I'll try to look into why the blank report issue happens. I think most likely my code here isn't working as intended.

https://github.com/Expensify/Expensify.cash/blob/910cbb4e4b728b3879dbf4bd7db1ebc7c3823ad2/src/libs/Navigation/AppNavigator/AuthScreens.js#L180-L188

Reply:
>Also, it seems that our navigation stack is not quite working as expected, because when at the top of navigation hierarchy (the Chat List page), pressing the native back button (or using the gesture) actually takes you deeper into the app by reopening the the 1:1 chat page which has been closed

@Julesssss the reason for this is because the Chats List page is essentially a drawer that's defaulted to visible, which is why hitting back "closes" the drawer and shows the reports (which is why that ios gesture behavior happens between the two screens as well).

@marcaaron I'm going to branch off later today and try reorganizing the stacks in a way that avoids this. I think it might be better if chats list wasnt a drawer and just the initial screen in the auth stack instead. the tricky part would be for larger devices which needs both the chat list and the reports as one "screen"

Reply:
Thanks, great discussion and points @Julesssss.

A couple of obvious solutions to **A** might be:

1. Disallow the the gesture that allows the drawer to pop open or closed on Android and keep the drawer navigation
2. Stop using a drawer navigator entirely and just make the sidebar the "root view" instead of a toggleable drawer

I found getting 2 to work correctly to be a bit of a challenge on web, but maybe we can try again. At some point, I had this set up differently, but went with the drawer since it allowed us to achieve the correct paneled view design on mWeb, web, desktop. I loosely followed the [official example](https://github.com/react-navigation/react-navigation/tree/main/example) (with some modifications) and setting the [`drawerType="permanent"`](https://reactnavigation.org/docs/drawer-navigator/#drawertype) gets us pretty close to our current design.

**B** sounds like a side effect of using the drawer navigator like @Maftalion has pointed out

Reply:
alright, figured out the report not appearing issue and resolved it, nice catch @Maftalion

Reply:
@Julesssss creating a new issue to investigate the back button points + feasibility of ditching the drawer navigator

https://github.com/Expensify/Expensify/issues/157238